File: src/data/services/IAITutorService.ts
import { LessonStep } from '../../domain/entities/Lesson';

export interface AIExplanationRequest {
  courseId: string;
  skillId: string;
  questionId: string;
  questionText: string;
  selectedOption: string;
  correctOption?: string;
  learnerName?: string;
}

export interface AIExplanationResponse {
  tutorName: string;
  title: string;
  explanation: string;
  remediationStep: string;
  suggestedAction: string;
  motivation?: string;
  reinforcementNeeded?: boolean;
  isDeterministicFallback: boolean;
}

export interface GenerateQuestionRequest {
  courseId: string;
  skillId: string;
  skillName?: string;
  difficulty: 'easy' | 'medium' | 'hard';
  learnerScore?: number;
  recentMistakes?: string[];
}

export interface GeneratedQuestionResponse {
  id: string;
  courseId: string;
  skillId: string;
  text: string;
  formulaLatex?: string | null;
  options: string[];
  difficulty: 'easy' | 'medium' | 'hard';
  isAiGenerated: boolean;
  isDeterministicFallback?: boolean;
}

export interface GenerateReinforcementRequest {
  courseId: string;
  skillId: string;
  skillName?: string;
  misconceptionTitle?: string;
  previousQuestionText?: string;
}

export interface GenerateLessonRequest {
  courseId: string;
  skillId: string;
  topic?: string;
  level?: string;
  difficulty?: 'easy' | 'medium' | 'hard' | number;
  questionCount?: number;
  language?: string;
  learnerScore?: number;
  recentMistakes?: string[];
}

export interface GeneratedLessonResponse {
  lessonId: string;
  courseId: string;
  skillId: string;
  topic: string;
  level: string;
  difficulty: 'easy' | 'medium' | 'hard' | string;
  title: string;
  summary: string;
  objective: string;
  estimatedMinutes: number;
  steps: LessonStep[];
  questions: GeneratedQuestionResponse[];
  isAiGenerated: boolean;
  model?: string;
  isDeterministicFallback?: boolean;
}

export interface IAITutorService {
  explainMistake(request: AIExplanationRequest): Promise<AIExplanationResponse>;
  generateQuestion?(request: GenerateQuestionRequest): Promise<GeneratedQuestionResponse>;
  generateReinforcement?(request: GenerateReinforcementRequest): Promise<GeneratedQuestionResponse>;
  generateLesson?(request: GenerateLessonRequest): Promise<GeneratedLessonResponse>;
}

File: mobile/lib/data/services/i_ai_tutor_service.dart
class AIExplanationRequest {
  final String courseId;
  final String skillId;
  final String questionId;
  final String questionText;
  final String selectedOption;
  final String correctOption;
  final String? learnerName;

  const AIExplanationRequest({
    required this.courseId,
    required this.skillId,
    required this.questionId,
    required this.questionText,
    required this.selectedOption,
    required this.correctOption,
    this.learnerName,
  });
}

class AIExplanationResponse {
  final String tutorName;
  final String title;
  final String explanation;
  final String remediationStep;
  final String suggestedAction;
  final bool isDeterministicFallback;

  const AIExplanationResponse({
    required this.tutorName,
    required this.title,
    required this.explanation,
    required this.remediationStep,
    required this.suggestedAction,
    this.isDeterministicFallback = true,
  });
}

class GenerateLessonRequest {
  final String courseId;
  final String skillId;
  final String? topic;
  final String? level;
  final String? difficulty;
  final int? questionCount;
  final int? learnerScore;

  const GenerateLessonRequest({
    required this.courseId,
    required this.skillId,
    this.topic,
    this.level,
    this.difficulty,
    this.questionCount,
    this.learnerScore,
  });
}

class GeneratedLessonResponse {
  final String lessonId;
  final String courseId;
  final String skillId;
  final String topic;
  final String level;
  final String difficulty;
  final String title;
  final String summary;
  final String objective;
  final int estimatedMinutes;
  final List<dynamic> steps;
  final List<dynamic> questions;
  final bool isAiGenerated;
  final String? model;

  const GeneratedLessonResponse({
    required this.lessonId,
    required this.courseId,
    required this.skillId,
    required this.topic,
    required this.level,
    required this.difficulty,
    required this.title,
    required this.summary,
    required this.objective,
    required this.estimatedMinutes,
    required this.steps,
    required this.questions,
    this.isAiGenerated = true,
    this.model,
  });
}

abstract class IAITutorService {
  Future<AIExplanationResponse> explainMistake(AIExplanationRequest request);
  Future<GeneratedLessonResponse> generateLesson(GenerateLessonRequest request);
}
